commit bf2560129d0b2d49f3f6082b97cb602da0435ccd (refs/changes/33/933/3)
Author: Bryan Cantrill <bryan@joyent.com>
Date:   2016-11-22T01:50:39+00:00 (2 years, 11 months ago)
    
    joyent/eng#11 bashstyle should detect trailing whitespace
    joyent/eng#12 bashstyle should be jsstyle-clean
    joyent/eng#13 update jsstyle
    Reviewed by: David Pacheco <dap@joyent.com>
    Approved by: David Pacheco <dap@joyent.com>

diff --git a/Makefile b/Makefile
index 835a1d2..1394830 100644
--- a/Makefile
+++ b/Makefile
@@ -86,7 +86,7 @@ include ./tools/mk/Makefile.defs
 #
 DOC_FILES	 = index.md boilerplateapi.md
 JSON_FILES	 = package.json
-JS_FILES	:= $(shell find lib test -name '*.js')
+JS_FILES	:= $(shell find lib test -name '*.js') tools/bashstyle
 JSL_FILES_NODE	 = $(JS_FILES)
 JSSTYLE_FILES	 = $(JS_FILES)
 
diff --git a/deps/jsstyle b/deps/jsstyle
index d75b7ca..9600c7e 160000
--- a/deps/jsstyle
+++ b/deps/jsstyle
@@ -1 +1 @@
-Subproject commit d75b7ca8308be17c80e2b120f2a01d4a0c20d8a8
+Subproject commit 9600c7e56c84f3a74d6e3d70c336e86f7a3e3769
diff --git a/tools/bashstyle b/tools/bashstyle
index 3e8052e..3ea6622 100755
--- a/tools/bashstyle
+++ b/tools/bashstyle
@@ -79,19 +79,22 @@ function checkFile(filename)
 		if (line.match(/# END BASHSTYLED/)) {
 			if (styled != true) {
 				nerrors++;
-				console.log('%s: %d: END BASHSTYLED w/o corresponding BEGIN',
-				            filename, i);
+				console.log('%s: %d: END BASHSTYLED ' +
+				    'w/o corresponding BEGIN', filename, i);
 			}
 			styled = false;
 		}
 
+		/*JSSTYLED*/
 		if (!styled && line.match(/^\s*local\s+(\w+)\s*=.*\$\(/)) {
 			nerrors++;
+			/*JSSTYLED*/
 			var m = line.match(/^\s*local\s+(\w+)\s*=/);
 			console.log('%s: %d: declaring and setting a "local" ' +
 				'var in the same statement ' +
 				'ignores a subshell return code ' +
-				'<http://www.tldp.org/LDP/abs/html/localvar.html#EXITVALANOMALY01>: ' +
+				'<http://www.tldp.org/LDP/abs/html/' +
+				'localvar.html#EXITVALANOMALY01>: ' +
 				'local %s=...',
 				filename, i, m[1]);
 		}
@@ -100,7 +103,8 @@ function checkFile(filename)
 		// groups before and after brackets to ease search/replace.
 		if (!styled && line.match(/(^|[^\[])\[(\s.+\s)\]([^\]])/)) {
 			nerrors++;
-			console.log('%s: %d: prefer [[ to [ for tests.', filename, i);
+			console.log('%s: %d: prefer [[ to [ for tests.',
+			    filename, i);
 		}
 
 		if (!styled && line.length > 80) {
@@ -108,6 +112,12 @@ function checkFile(filename)
 			console.log('%s: %d: line exceeds 80 columns',
 			    filename, i);
 		}
+
+		if (!styled && line.match(/\s+$/)) {
+			nerrors++;
+			console.log('%s: %d: line ends in whitespace',
+			    filename, i);
+		}
 	}
 
 	if (styled) {
@@ -118,7 +128,8 @@ function checkFile(filename)
 
 
 	/*
-	 * No sane editor lets you save a file without a newline at the very end.
+	 * No sane editor lets you save a file without a newline at the
+	 * very end.
 	 */
 	if (lines[lines.length - 1].length !== 0) {
 		nerrors++;
@@ -155,7 +166,7 @@ function expandTabs(text)
 
 		do {
 			out += ' ';
-		}  while (--k > 0);
+		} while (--k > 0);
 
 		col += k;
 	}
diff --git a/tools/jsstyle.conf b/tools/jsstyle.conf
index bbf409a..0fcd409 100644
--- a/tools/jsstyle.conf
+++ b/tools/jsstyle.conf
@@ -8,7 +8,6 @@
 # Copyright (c) 2014, Joyent, Inc.
 #
 
-indent=2
 doxygen
 unparenthesized-return=0
 blank-after-start-comment=0
